Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/nbclient/client.py", line 1269, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/nbclient/util.py", line 85, in wrapped
    return just_run(coro(*args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/nbclient/util.py", line 60, in just_run
    return loop.run_until_complete(coro)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/nbclient/client.py", line 707, in async_execute
    await self.async_execute_cell(
  File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/nbclient/client.py", line 1025, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/opt/hostedtoolcache/Python/3.11.7/x64/lib/python3.11/site-packages/nbclient/client.py", line 919, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# refer to the shape by its URI
app1_shape = URIRef("urn:simpleapp/app1")

print(sc.shape_to_query(app1_shape))
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mTypeError[0m                                 Traceback (most recent call last)
Cell [0;32mIn[4], line 4[0m
[1;32m      1[0m [38;5;66;03m# refer to the shape by its URI[39;00m
[1;32m      2[0m app1_shape [38;5;241m=[39m URIRef([38;5;124m"[39m[38;5;124murn:simpleapp/app1[39m[38;5;124m"[39m)
[0;32m----> 4[0m [38;5;28mprint[39m([43msc[49m[38;5;241;43m.[39;49m[43mshape_to_query[49m[43m([49m[43mapp1_shape[49m[43m)[49m)

File [0;32m~/work/BuildingMOTIF/BuildingMOTIF/buildingmotif/dataclasses/shape_collection.py:270[0m, in [0;36mShapeCollection.shape_to_query[0;34m(self, shape)[0m
[1;32m    259[0m     [38;5;28;01mdef[39;00m [38;5;21mshape_to_query[39m([38;5;28mself[39m, shape: URIRef) [38;5;241m-[39m[38;5;241m>[39m [38;5;28mstr[39m:
[1;32m    260[0m [38;5;250m        [39m[38;5;124;03m"""[39;00m
[1;32m    261[0m [38;5;124;03m        This method takes a URI representing a SHACL shape as an argument and returns[39;00m
[1;32m    262[0m [38;5;124;03m        a SPARQL query selecting the information which would be used to satisfy that[39;00m
[0;32m   (...)[0m
[1;32m    268[0m [38;5;124;03m            ?target <path> <value>[39;00m
[1;32m    269[0m [38;5;124;03m        """[39;00m
[0;32m--> 270[0m         clauses, project [38;5;241m=[39m [43m_shape_to_where[49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mgraph[49m[43m,[49m[43m [49m[43mshape[49m[43m)[49m
[1;32m    271[0m         preamble [38;5;241m=[39m [38;5;124m"""[39m[38;5;124mPREFIX sh: <http://www.w3.org/ns/shacl#>[39m
[1;32m    272[0m [38;5;124mPREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>[39m
[1;32m    273[0m [38;5;124mPREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>[39m
[1;32m    274[0m [38;5;124m        [39m[38;5;124m"""[39m
[1;32m    275[0m         [38;5;28;01mreturn[39;00m [38;5;124mf[39m[38;5;124m"[39m[38;5;132;01m{[39;00mpreamble[38;5;132;01m}[39;00m[38;5;124m SELECT [39m[38;5;132;01m{[39;00m[38;5;124m'[39m[38;5;124m [39m[38;5;124m'[39m[38;5;241m.[39mjoin(project)[38;5;132;01m}[39;00m[38;5;124m WHERE [39m[38;5;130;01m{{[39;00m[38;5;130;01m\n[39;00m[38;5;132;01m{[39;00mclauses[38;5;132;01m}[39;00m[38;5;130;01m\n[39;00m[38;5;130;01m}}[39;00m[38;5;124m"[39m

File [0;32m~/work/BuildingMOTIF/BuildingMOTIF/buildingmotif/dataclasses/shape_collection.py:340[0m, in [0;36m_shape_to_where[0;34m(graph, shape)[0m
[1;32m    338[0m [38;5;66;03m# handle targetNode[39;00m
[1;32m    339[0m targetNode [38;5;241m=[39m graph[38;5;241m.[39mobjects(shape, SH[38;5;241m.[39mtargetNode)
[0;32m--> 340[0m [38;5;28;01mif[39;00m [38;5;28;43mlen[39;49m[43m([49m[43mtargetNode[49m[43m)[49m [38;5;241m>[39m [38;5;241m1[39m:
[1;32m    341[0m     [38;5;28;01mraise[39;00m [38;5;167;01mValueError[39;00m(
[1;32m    342[0m         [38;5;124m"[39m[38;5;124mCannot currently have more than one sh:targetNode in shacl-to-sparql conversion[39m[38;5;124m"[39m
[1;32m    343[0m     )
[1;32m    344[0m [38;5;28;01mif[39;00m targetNode:

[0;31mTypeError[0m: object of type 'generator' has no len()
TypeError: object of type 'generator' has no len()

